### PR TITLE
Feature/page meta data

### PR DESCRIPTION
**Pull request by apoorvaasangi94**

### CODE REVIEW

File: bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+     * fetchTagTitle method is used to fetch the tag title of the tag on which the page is tagged
+     */
+
+    @PostConstruct
+    private void fetchTagTitle() {
+        if (resource != null) {
+            LOG.debug("Flow of control has entered inside fetchTagTitle method");
+            TagManager tagManager = resource.getResourceResolver().adaptTo(TagManager.class);
+            String[] mappedTags = resource.getValueMap().get("cq:tags", String[].class);
+            for (String tags : mappedTags) {
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java]: Tags shouldn't be plural

File: bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                Tag tag = tagManager.resolve(tags);
+                tagTitles.add(tag.getTitle());
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java]: What happens if you delete tag from tagging console after it is tagged?

File: bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     */
+
+    @PostConstruct
+    private void fetchTagTitle() {
+        if (resource != null) {
+            LOG.debug("Flow of control has entered inside fetchTagTitle method");
+            TagManager tagManager = resource.getResourceResolver().adaptTo(TagManager.class);
+            String[] mappedTags = resource.getValueMap().get("cq:tags", String[].class);
+            for (String tags : mappedTags) {
+                Tag tag = tagManager.resolve(tags);
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java]: Null not handled

File: bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/structure/base-page/head.html [hunk 4]
```diff
@@ -1,7 +1,12 @@
-<head>
+<head data-sly-use.result="com.bootcamp.core.models.PageMetaData">
     <sly data-sly-include="/libs/wcm/core/components/init/init.jsp" ></sly>
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>${pageProperties.jcr:title}</title>
+    <meta name="keywords" content="${result.tagTitles}"/>
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/structure/base-page/head.html]: What happens if there are no tags at all?

File: bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import javax.annotation.PostConstruct;
+import java.util.*;
+
+/**
+ * This class is based on Resource sling model and
+ * this class is used to fetch the values for keywords for meta property
+ * in head section and class is invoked in head section of the page
+ * Creating list variable to store the titles of the tag
+ * To invoke this model in slightly use data-sly-use
+ * For reference data-sly-use.result="${'com.bootcamp.core.models.PageMetaData'}"
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java]: Your doc and the way you have invoked don't match

File: bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+ * In above line we are invoking the model class and storing it in variable result.
+ * For accessing the list use result.tagTitles
+ */
+@Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java]: What resource will be used here?

Comment (apoorvaasangi94) [reply bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java]: Resource will be our page

File: bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+     */
+
+    @PostConstruct
+    private void fetchTagTitle() {
+        if (resource != null) {
+            LOG.debug("Flow of control has entered inside fetchTagTitle method");
+            TagManager tagManager = resource.getResourceResolver().adaptTo(TagManager.class);
+            String[] mappedTags = resource.getValueMap().get("cq:tags", String[].class);
+            if (mappedTags != null) {
+                for (String tag : mappedTags) {
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java]: Add this logic into a different method and invoke it here. Give proper method names

Comment (apoorvaasangi94) [reply bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java]: yes i have made this change

File: bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+    }
+
+    /**
+     * Fetch Tag Titles method is used to fetch the tag title of the tag on which the page is tagged
+     * @param mappedTags
+     * @param tagManager
+    */
+    private void fetchTagTitles(TagManager tagManager, String[] mappedTags) {
+        if (mappedTags != null) {
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/core/src/main/java/com/bootcamp/core/models/PageMetaData.java]: This condition is incorrect

**Pull request merged**